Dockerfile

```diff
@@ -32,5 +24,15 @@ RUN apt update && apt install --no-install-recommends -y \
 
 COPY --from=build /code /code
 WORKDIR /code
+
+# Create a non-root user and group
+RUN addgroup --system appgroup && adduser --system --ingroup appgroup appuser
+
+# Change ownership of the files to the non-root user
+RUN chown -R appuser:appgroup /code
```

Comment:
This can be done directly when copying, using the `--chown` parameter of `COPY` (see https://docs.docker.com/reference/dockerfile/#copy---chown---chmod ).
It will speed up the build! :)